Kickstarter-iOS/Views/Controllers/BetaToolsViewController.swift

```diff
@@ -155,13 +210,24 @@ internal final class BetaToolsViewController: UIViewController {
     self.present(alert, animated: true, completion: nil)
   }
 
-  private func languageDidChange(language: Language) {
+  private func updateLanguage(language: Language) {
     AppEnvironment.updateLanguage(language)
+
     NotificationCenter.default.post(
       name: Notification.Name.ksr_userLocalePreferencesChanged,
       object: nil,
       userInfo: nil
     )
+
+    self.navigationController?.popViewController(animated: true)
+  }
+
+  private func updateEnvironment(environment: EnvironmentType) {
+    let serverConfig = ServerConfig.config(for: environment)
```

Comment:
Just one thing to bear in mind when doing that sort of thing inside the view model is - if the environment is involved in the chain of signals that perform that replacement you may run the risk of the environment being switched out from underneath and the subscriptions being lost. Little tricky to explain but I had this happen once before something like:

```swift
AppEnvironment.current.signal
  .map { // transform something... }
  .observeValues { // replace environment }
```
Because this depends on the current environment doing the replacement here can be problematic. This might not exactly apply in what @Scollaco is suggesting 😄 My feeling is to do the replacement outside of the view model as it is now.

Reply:
Yeah I thought about this and my feeling is that actually updating the environment is more of a side effect. This way, we can be certain that the right `EnvironmentType` is being sent, and we've separately tested that side-effect behavior works as expected.

We also do it this way in [`AppDelegate`](https://github.com/kickstarter/ios-oss/blob/master/Kickstarter-iOS/AppDelegate.swift#L68)